### PR TITLE
[no ticket] update PR template

### PR DESCRIPTION
**Pull request by Qi77Qi**
\<your comments for this PR go here\>

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -5,11 +5,6 @@ Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/d
...
-- [ ] I've documented my API changes in Swagger
-
-In all cases:
-
+- [ ] I've documented my API changes in Swagger. Bump `version` field in [API doc](src/main/resources/swagger/api-docs.yaml).
 - [ ] Get a thumbsworth of review and PO signoff if necessary
-- [ ] Verify all tests go green
-- [ ] Squash and merge; Delete your branch after this
-- [ ] Test this change deployed correctly and works on dev environment after deployment
+- [ ] In most cases, keep your pull request branch up-to-date before merging; You may by-pass this with admin privilege if the change is trivial.
```
Comment (kyuksel) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: Should we clarify what `trivial` means?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -5,11 +5,6 @@ Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/d
...
 
-- [ ] I've documented my API changes in Swagger
-
-In all cases:
-
+- [ ] I've documented my API changes in Swagger. Bump `version` field in [API doc](src/main/resources/swagger/api-docs.yaml).
 - [ ] Get a thumbsworth of review and PO signoff if necessary
-- [ ] Verify all tests go green
-- [ ] Squash and merge; Delete your branch after this
-- [ ] Test this change deployed correctly and works on dev environment after deployment
```
Comment (kyuksel) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: `dev deploy` sometimes fails, which makes me wonder if we should still say something like 

> verify that the `dev deploy` job triggered by the PR merging succeeds

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (kyuksel) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: Also, should all Callistans have admin privilege?

Comment (jdcanas) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: Hmm... why do we want to allow people to not rebase on small changes? Do you mean they don't have to wait for the tests if its a trivial change?

If that's the case, I think `plain text change` is a better descriptor than trivial. 

Do you mean not waiting for the checks to go green? I think its worth keeping something similar to line 13, i.e. `Verify all checks pass`

Sorry for being pedantic, but I think this checklist is pretty important for new contributors/team members

**Pull request closed without merging**